### ISSUE TITLE
.NET 7 issues

### ISSUE THREAD

**Issue by poo-pee-poo** [kant2002/WinFormsComInterop#38]
Hello,
Thank you for making this great repository!

When using .NET 7 Preview 5, I am experiencing some issues with the `SampleWindowsForms` project.
It appears in `Program.cs`, there's a `#if` disabling `ComWrappers` when running .NET 7:
```c#
#if !NET7_0_OR_GREATER
            ComWrappers.RegisterForMarshalling(WinFormsComInterop.WinFormsComWrappers.Instance);
#endif
```

It seems disabling `ComWrappers` causes the error message "Built-in COM has been disabled via a feature switch." on application launch. However, by removing the `#if !NET7_0_OR_GREATER` statement, this issue goes away. My question is, why disable `ComWrappers` registration on .NET 7?

The last issue I experience is with the windows forms **designer**. When double clicking on for example, `MainForm.cs`, numerous error messages pop up. One of them being "The control System.Windows.Forms.TextBox has thrown an unhandled exception in the designer and has been disabled."
**This issue does not appear when running the app normally (via Ctrl + F5), but only in the designer**.
I am assuming that this is because when running the app normally, `ComWrappers.RegisterForMarshalling` is able to be called. However, when in the designer, `ComWrappers.RegisterForMarshalling` cannot be called.

However, most of the designer works. It's just `TextBox`, and `NumericUpDown` that does not work and shows an error.
I was wondering, is there any way around this problem?

Thanks

**Comment by kant2002** [comment]
I plan to get rid of built-in COM in WinForms so this library would be needed for net6.0 only. I would say it almost working except Accessibility support where I hit some issues

**Comment by poo-pee-poo** [comment]
@kant2002 Ah ok, I see. I think the major missing parts are for `TextBox` and `NumericUpDown`, without built-in COM support, correct?

Also, I was wondering if you had any advice for the WinForms designer support? Like is there a way to get the WinForms designer to load the `ComWrappers`?

Thank you for this great library!

**Comment by kant2002** [comment]
If you want designer you for example do not include call to `ComWrappers.RegisterForMarshalling`. That way debug build run with designer but release when publish with NativeAOT will work also. And obviously test test test final binary with NativeAOT

**Comment by poo-pee-poo** [comment]
@kant2002 Thank you for the response!
However, I'm not quite sure I understand what you're saying. Currently I don't have the Windows forms designer calling `RegisterForMarshalling`. Its just when I double click on the Form, and visual studio opens the designer, an error message appears in `TextBoxBase` saying that built-in COM marshalling has been disabled.

I was just wondering if I could somehow make the Visual Studio winforms designer load the custom `ComWrappers`.
Thank you!

**Comment by kant2002** [comment]
What I'm saying @poo-pee-poo is that to make designer works with this library is probably imposible, so I would not recommend try to do that. Maybe you can create `[ModuleInitializer]` in your assembly, and that will do the trick, but risk is not worth it IMO.

I suggest you use designer without `ComWrappers.RegisterForMarshalling` augmentation, and use it only for Release configuration. Or completely ditch designer (which you probably do not want).

**Comment by poo-pee-poo** [comment]
Thank you for the reply @kant2002!
Apologies for dropping off the radar!

I ended up adding `Condition`'s in the project file to conditionally include `WinFormsComInterop` and `ILCompiler`.

This worked great, thank you for the advice!

One concern I have is in order to get the project to publish, I had to remove
```
<IlcCompileInput Include="--reflectedonly" />
```
in `SampleWindowsForms`, because an error message kept popping up saying that was an unknown option.

I was wondering if removing `reflectedonly` could have any adverse affects.
Thank you!
